backend/routes/user.js

```diff
@@ -318,9 +318,8 @@ user.post("/goal", async (req, res) => {
         description,
         targetAmount: parseFloat(targetAmount),
         currentAmount: parseFloat(currentAmount),
-        deadline,
+        deadline: new Date(deadline),
         date: new Date(deadline),
```

Comment:
Are both deadline and date supposed to be set to the same value? If date is meant to represent the creation date, it should be `new Date()` instead.

```suggestion
        deadline: new Date(deadline),
        date: new Date(),
```